### PR TITLE
Add Sprayproxy ADR

### PR DESCRIPTION
**Pull request by enarha**
Added the Sprayproxy ADR. Also removed few Sprayproxy details initially documented as part of the pipeline-service-via-operator ADR, so less duplication and easier management.

### CODE REVIEW

File: ADR/0009-pipeline-service-via-operator.md [hunk 1]
```diff
@@ -48,9 +48,7 @@ the OSP CRDs. The following changes are specific to RHTAP:
 Furthermore, as the service will be accessed through CodeReadyToolchain (CRT), the
 following changes are also specific to RHTAP:
 - Deploying a proxy (known as `SprayProxy`) on the CRT host cluster that redirects
```
Comment (Roming22) [inline ADR/0009-pipeline-service-via-operator.md:50]: Do you think it'd be useful to have a link to the new ADR here?

File: ADR/0031-sprayproxy.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# 0031 Sprayproxy
+
+Created: 2023-10-24
+
+## Status
+
+Proposed
```
Comment (Roming22) [inline ADR/0031-sprayproxy.md:7]: Since we're documenting the current implementation, I would put it as 'Accepted'.

File: ADR/0031-sprayproxy.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+## Decision
+
+Deploy a service (`Sprayproxy`) on the AppStudio host (frontend) clusters. The service route is configured in the GitHub App as a `Webhook URL`, so all webhook requests are directed to it. The service has a list of backends configured. The service does not distinguish between the type of requests the way PaC does (pull-request/push/comment etc), it treats them all equally. For each incoming request, a new outgoing request is constructed with the original payload and destination of each of the member clusters.
+
+The service performs the following checks on incoming requests:
+
+- Validating the request is from GitHub using a shared secret. The process is officially documented [here](https://docs.github.com/en/webhooks/using-webhooks/validating-webhook-deliveries).
+- [Payloads are capped at 25 MB](https://docs.github.com/en/webhooks/webhook-events-and-payloads#payload-cap), but that's configurable.
+
+The QE team requires the ability to dynamically add/remove backends. That functionality is disabled by default as it represents a security risk.
```
Comment (Roming22) [inline ADR/0031-sprayproxy.md:22]: I propose to expand this paragraph a little.
1. Explain where the security risks come from (secret to access the repo is part of payload, bad actor could access private content if they get the forwarded payload).
2. Explain a little bit on how the registration/unregistration mechanism works.

**Comment by Roming22** [comment]
There's a typo in the commit message 'ARD' -> 'ADR'.

File: ADR/0009-pipeline-service-via-operator.md (hunk 1 above)
Comment (enarha) [reply ADR/0009-pipeline-service-via-operator.md:50]: My feeling was these has to sort of self contained, but I saw such links in other ADRs, so added the link as suggested.

File: ADR/0031-sprayproxy.md (hunk 2 above)
Comment (enarha) [reply ADR/0031-sprayproxy.md]: Agree. That was my feeling as well. Updated.

File: ADR/0031-sprayproxy.md (hunk 3 above)
Comment (enarha) [reply ADR/0031-sprayproxy.md]: Added.

**Comment by enarha** [comment]
> There's a typo in the commit message 'ARD' -> 'ADR' (x2).

Yea, I made that typo 200 times. For some reason "ADR" doe snot come naturally to me.

**Review by Roming22 — APPROVED** [review]

**Comment by Roming22** [comment]
@adambkaplan  @sbose78  Can you please review. If it looks good, please approve and merge.

**Review by adambkaplan — APPROVED** [review]
LGTM

**Pull request merged**